lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -229,10 +229,11 @@ def set_target_dependencies
               aggregate_target.native_target.add_dependency(pod_target.native_target)
               configure_app_extension_api_only_for_target(pod_target) if is_app_extension
 
+              add_dependent_targets_to_native_target(pod_target.dependent_targets,
```

Comment:
is there a test that covers this case?

Reply:
I haven't been able to figure out how the tests in `pods_project_generator_spec.rb` do validation to test Xcode project contents. 

Do you have any guidance?

Reply:
Ah yes I remember now this is a bit difficult to write but doable.

Look into:
```ruby
          describe '#set_test_target_dependencies' do
            ...
          end
```
Most of it is mocked and ensures all mocks receive the correct method calls with parameters.

Reply:
Thanks! That helped.